# robot_id: 0
robot_num: 2 # desired number of robots in the network
sensor_input_mode: 1 # 0: depth image (default) 1: pointcloud 
localization_mode: 1 # 0: pose (default) 1: odom
depth_image_topic: /depth
# point_cloud_topic: /theia/os_cloud_node/points
# point_cloud_topic: /camera/depth/points
# pose_topic: /ugv/mavros/local_position/pose
# odom_topic: /vins_estimator/odometry
odom_topic: /odom
point_cloud_topic: /point_cloud

# robot size
robot_size: [1.1, 0.8, 0.8] #1.15, 1.05, 0.0

# Camera Parameters
# depth_intrinsics: [554.254691191187, 554.254691191187, 320.5, 240.5] # fx,  fy, cx, cy
depth_intrinsics: [408.5106506347656, 229.78724670410156, 640.0, 360.0] # isaac sim jetbot
depth_scale_factor: 1000 # 1000 for Intel Realsense Camera
depth_min_value: 0.5
depth_max_value: 5.0
depth_filter_margin: 2 # filter
depth_skip_pixel: 2 # filter
# image_cols: 640
# image_rows: 480

# jetbot camera isaac sim 
image_cols: 1280
image_rows: 720
# body_to_camera: [1.0,  0.0,  0.0,  0.00,
#                   0.0,  1.0,  0.0,  0.0 ,   
#                   0.0,  0.0,  1.0,  0.43,
#                   0.0,  0.0,  0.0,  1.0] # or body to lidar

# jetbot
# body_to_camera: [0.0,  0.0,  1.0,  0.068,
#                 -1.0,  0.0,  0.0,  0.0 ,   
#                  0.0, -1.0,  0.0,  0.04,
#                  0.0,  0.0,  0.0,  1.0]

# os_lidar isaac sim
body_to_camera: [1.0,  0.0,  0.0,  0.0,
                 0.0,  1.0,  0.0,  0.0,   
                 0.0,  0.0,  1.0,  0.4,
                 0.0,  0.0,  0.0,  1.0]

# map in this robot to global map. For multi-robot map transimission
global_to_map_1: [1.0,  0.0,  0.0,  0.0,
                  0.0,  1.0,  0.0,  0.0,   
                  0.0,  0.0,  1.0,  0.0,
                  0.0,  0.0,  0.0,  1.0]
global_to_map_2: [1.0,  0.0,  0.0, -2.0,
                  0.0,  1.0,  0.0, -2.0,   
                  0.0,  0.0,  1.0,  0.0,
                  0.0,  0.0,  0.0,  1.0]

# Raycasting
raycast_max_length: 5.0
p_hit: 0.70
p_miss: 0.35
p_min: 0.12
p_max: 0.97
p_occ: 0.80


# Map
map_resolution: 0.1
ground_height: -0.7 # m
map_size: [25, 25, 4] # meter. in x y z direction (reserved size)
local_update_range: [5, 5, 4]
local_bound_inflation: 3.0 # inflate local bound in meter
clean_local_map: false
preload_map_dir: /home/cerlab-ugv/catkin_ws/static_map.pcd

# visualziation
local_map_size: [25, 15, 6] # meter. in x y z direction (only for visualization)
max_height_visualization: 3.0 # m
visualize_global_map: true
verbose: false
